src/hooks/v1/RelaunchV1ViaV3Create.ts

```diff
@@ -0,0 +1,148 @@
+import { SECONDS_IN_DAY } from 'constants/numbers'
+import { ProjectMetadataContext } from 'contexts/projectMetadataContext'
+import { V1ProjectContext } from 'contexts/v1/projectContext'
+import { PayoutMod, TicketMod } from 'models/mods'
+import { consolidateMetadata } from 'models/project-metadata'
+import { useRouter } from 'next/router'
+import { useCallback, useContext, useMemo } from 'react'
+import {
+  permyriadToPercent,
+  percentToPermyriad,
+  perbicentToPercent,
+  fromWad,
+} from 'utils/format/formatNumber'
+import { decodeFundingCycleMetadata } from 'utils/v1/fundingCycle'
+import { splitPercentFrom, discountRateFrom } from 'utils/v2v3/math'
+
+export const useRelaunchV1ViaV3Create = () => {
+  const router = useRouter()
+
+  const { projectMetadata } = useContext(ProjectMetadataContext)
+  const { queuedFC, queuedPayoutMods, queuedTicketMods } =
+    useContext(V1ProjectContext)
+  const fundingCycleMetadata = decodeFundingCycleMetadata(queuedFC?.metadata)
+
+  const isReady = useMemo(() => {
+    return !(
+      projectMetadata === undefined ||
+      queuedFC === undefined ||
+      queuedPayoutMods === undefined ||
+      queuedTicketMods === undefined
+    )
+  }, [projectMetadata, queuedFC, queuedPayoutMods, queuedTicketMods])
+
+  const v1ConvertedInitialState = useMemo(() => {
+    // No idea why we have to do this way but it is what it is.
+    const convertPayoutMods = (mods: PayoutMod[]) => {
```

Comment:
**suggestion** my bad, we actually do have a fn for this that might be helpful: `toSplit` https://github.com/jbx-protocol/juice-interface/blob/93dcc2feff8fb57632d112ab313abf0379a7af8a/src/utils/splits.ts#L16

Reply:
The formats are a lil different so im going to opt to leave this one as it should hopefully not be long lived code